tests/test_extract.py

```diff
@@ -131,6 +131,45 @@ def helperExtractCache(self, test_name, hpss_path, zstash_path=ZSTASH_PATH):
             error_message = 'The zstash cache does not contain expected files.\nIt has: {}'.format(files)
             self.stop(error_message)
 
+    def helperExtractWildcard(self, test_name, hpss_path, zstash_path=ZSTASH_PATH):
+        """
+        Test `zstash extract` with wildcard.
+        """
+        self.assertWorkspace()
+        print_in_box(test_name)
+        self.hpss_path = hpss_path
+        if self.hpss_path.lower() == 'none':
+            use_hpss = False
+        else:
+            use_hpss = True
+        os.mkdir(self.test_dir)
+        os.chdir(self.test_dir)
+        write_file('aa0.txt', 'aa0')
+        write_file('aa1.txt', 'aa1')
+        write_file('ab2.txt', 'ab2')
+        write_file('ab3.txt', 'ab3')
+        os.chdir(TOP_LEVEL)
+        self.create(use_hpss, zstash_path)
+        # Rename test_dir to backup_dir so that the new test_dir won't have the files we're trying to extract.
+        os.rename(self.test_dir, self.backup_dir)
+        os.mkdir(self.test_dir)
+        os.chdir(self.test_dir)
+        if not use_hpss:
+            shutil.copytree('{}/{}/{}'.format(TOP_LEVEL, self.backup_dir, self.cache), self.copy_dir)
+        cmd = '{}zstash extract --hpss={} ab*'.format(zstash_path, self.hpss_path)
+        output, err = run_cmd(cmd)
+        print(output + err)
+        expected_present = ['Extracting ab2.txt', 'Extracting ab3.txt']
+        self.check_strings(cmd, output + err, expected_present, [])
+        cmd = '{}zstash extract --hpss={} a*'.format(zstash_path, self.hpss_path)
```

Comment:
Interestingly, adding double quotes seems to cause a failure and not adding them allows the tests to pass -- the reverse of our experience as users. I believe this is because `subprocess.Popen` requires `cmd` to be converted to a list and I imagine the conversion encapsulates `a*` in quotes. Not sure if there's a way to automate a test in such a way as to use the quotes.

Reply:
Try using `shell=True`? If that doesn't work, perhaps we should just close #96 without adding automated testing.